### ISSUE TITLE
[bug] Listener on the controller's "cluster" endpoint crashes if it is sent an unexpected request

### ISSUE THREAD

**Issue by justenwalker** [hashicorp/boundary#2072]
**Describe the bug**

Listener on the controller's "cluster" endpoint crashes if it is sent an unexpected request

**To Reproduce**

1. run `boundary dev` to set up a local boundary instance
2. Validate that boundary is listening on all the ports
   ```
    $ netstat -an -p tcp | grep LISTEN | grep 920
    tcp4       0      0  127.0.0.1.9202         *.*                    LISTEN
    tcp4       0      0  127.0.0.1.9203         *.*                    LISTEN
    tcp4       0      0  127.0.0.1.9201         *.*                    LISTEN
    tcp4       0      0  127.0.0.1.9200         *.*                    LISTEN
    ```
3. run `curl http://localhost:9201` to send it an unexpected request
4. Validate that boundary is no longer listening on 9201 (the "cluster" endpoint)
   ```
    $ netstat -an -p tcp | grep LISTEN | grep 920
    tcp4       0      0  127.0.0.1.9202         *.*                    LISTEN
    tcp4       0      0  127.0.0.1.9203         *.*                    LISTEN
    tcp4       0      0  127.0.0.1.9200         *.*                    LISTEN
    ```

**Expected behavior**

boundary should gracefully reject malformed packets on 9201 and continue to accept connections

**Additional context**

This seems to happen after the v0.8.0 upgrade

```
$ boundary version

Version information:
  Git Revision:        bc86b4e9e9adfdf07183cbf140a2802fb453b35f
  Version Number:      0.8.0
```

It is not reproduceable on `v0.7.6`

Also, nothing seems to get logged when this happens. I did not see any new output from the logstream after sending the curl.

**Comment by AdamBouhmad** [comment]
Thanks for reporting @justenwalker! We're looking into this currently.

**Comment by hsimon-hashicorp** [comment]
Thank you for the report, @justenwalker! This is fixed in #2073 and will be included in the next release. We really appreciate your contribution. Please feel free to open a new issue for any other potential regressions, bugs, oddities, or enhancement requests!
